python/ray/_private/resource_spec.py

```diff
@@ -276,6 +262,97 @@ def resolve(self, is_head: bool, node_ip_address: Optional[str] = None):
         return spec
 
 
+def _get_cuda_info(num_gpus):
+    """ Attemp to process the number and type of GPUs
+        Notice:
+            If gpu id not specified in CUDA_VISIBLE_DEVICES,
+            and num_gpus is defined in task or actor,
+            this function will return the input num_gpus, not 0
+
+    Returns:
+        (num_gpus, gpu_types)
+    """
+    gpu_ids = ray._private.utils.get_cuda_visible_devices()
+    # Check that the number of GPUs that the raylet wants doesn't
+    # exceed the amount allowed by CUDA_VISIBLE_DEVICES.
+    if num_gpus is not None and gpu_ids is not None and num_gpus > len(gpu_ids):
+        raise ValueError(
+                "Attempting to start raylet with {} GPUs, "
+                "but CUDA_VISIBLE_DEVICES contains {}.".format(num_gpus, gpu_ids)
+                )
+    if num_gpus is None:
+        # Try to automatically detect the number of GPUs.
+        num_gpus = _autodetect_num_gpus()
+        # Don't use more GPUs than allowed by CUDA_VISIBLE_DEVICES.
+        if gpu_ids is not None:
+            num_gpus = min(num_gpus, len(gpu_ids))
+
+    gpu_types = ""
+    try:
+        if importlib.util.find_spec("GPUtil") is not None:
+            gpu_types = _get_gpu_types_gputil()
+        else:
+            info_string = _get_gpu_info_string()
+            gpu_types = _constraints_from_gpu_info(info_string)
+    except Exception:
+        logger.exception("Could not parse gpu information.")
+
+    return num_gpus, gpu_types
+
+
+def _get_xpu_info(num_xpus):
+    """Attempt to process the number of XPUs as GPUs
+
+    XPU detect need oneAPI runtime. Here we enumrate XPU device by two method:
+      1. by using dpctl python API
+      2. by calling system commmand 'sycl-ls'
+
+    Notice that ONEAPI_DEVICE_SELECTOR environment variable should be unset
+    Or dpctl return filtered device by ONEAPI_DEVICE_SELECTOR
+
+    Returns:
+        The number of XPUs that detected by dpctl or oneAPI with specific backend and device type
+    """
+    xpu_ids = ray._private.utils.get_xpu_visible_devices()
+    if num_xpus is not None and xpu_ids is not None and num_xpus > len(xpu_ids):
+        raise ValueError(
+                "Attempting to start raylet with {} XPUs, "
+                "but XPU_VISIBLE_DEVICES contains {}.".format(num_xpus, xpu_ids)
+                )
+    if num_xpus is None:
+        has_dpctl = True
+        backend = ray_constants.RAY_DEVICE_XPU_BACKEND_TYPE
+        device_type = ray_constants.RAY_DEVICE_XPU_DEVICE_TYPE
+        try:
+            import dpctl
```

Comment:
@abhilash1910 I think this PR is to resolve general XPU support rather than Pytorch XPU support. Need a lightweight way to just detect XPU number and set environment variables.

Reply:
@xwu99  makes sense. LGTM ; further support can be added over this PR . Thanks